frontend/src/pages/beheer/gebouwen.jsx

```diff
@@ -1,235 +1,34 @@
-    setBuildingList(buildings);
-    setSearchResults(buildings);
-    await updateBuildingSelection(buildings[0]?.url);
-  };
-
-  const loadRegions = async () => {
-    const regions = await regionService.get();
-    setRegionList(regions);
-  };
-
-  const formatDate = (date) => {
-    let month = "" + (date.getMonth() + 1);
-    let day = "" + date.getDate();
-    const year = date.getFullYear();
-
-    if (month.length < 2) month = "0" + month;
-    if (day.length < 2) day = "0" + day;
-
-    return [year, month, day].join("-");
-  };
-
-  const loadPhotos = async (
-    url,
-    startDate = selectionStartDate,
-    endDate = selectionEndDate
-  ) => {
-    const photos = await buildingService.getPhotosByUrl(
-      url,
-      formatDate(startDate),
-      formatDate(endDate)
-    );
-    setPhotos(photos);
-  };
-
-  const loadComments = async (
-    url,
-    startDate = selectionStartDate,
-    endDate = selectionEndDate
-  ) => {
-    const comments = await buildingService.getCommentsByUrl(
-      url,
-      formatDate(startDate),
-      formatDate(endDate)
-    );
-    setComments(comments);
-  };
-
   useEffect(() => {
-    loadBuildings();
-    loadRegions();
-  }, []);
-
-  const filter = (params = {}) => {
-    const regionFilter =
-      "regions" in params ? params["regions"] : selectedRegions;
-    const searchFilter = "search" in params ? params["search"] : searchString;
-    const sortSetting = "sort" in params ? params["sort"] : sortingSetting;
-
-    let filterResults = buildingList;
-
-    if (regionFilter.length) {
-      filterResults = filterResults.filter((building) =>
-        JSON.stringify(regionFilter).includes(
-          JSON.stringify(building["region_name"])
-        )
-      );
-    }
-
-    filterResults = filterResults.filter((building) => {
-      const searchLower = searchFilter.current.value.toLowerCase();
-      return (
-        building["nickname"].toLowerCase().includes(searchLower) ||
-        building["description"].toLowerCase().includes(searchLower) ||
-        (building["address_line_1"] + " " + building["address_line_2"])
-          .toLowerCase()
-          .includes(searchLower)
-      );
-    });
-
-    if (sortSetting !== "") {
-      filterResults.sort((building_1, building_2) => {
-        switch (sortSetting) {
-          case "Naam":
-            return building_1["nickname"].toLowerCase() >
-              building_2["nickname"].toLowerCase()
-              ? 1
-              : -1;
-          case "Adres":
-            return (
-              building_1["address_line_1"] +
-              " " +
-              building_1["address_line_2"].toLowerCase()
-            ).toLowerCase() >
-              (
-                building_2["address_line_1"].toLowerCase() +
-                " " +
-                building_2["address_line_2"].toLowerCase()
-              ).toLowerCase()
-              ? 1
-              : -1;
-        }
-      });
-    }
+    const redirectToBuilding = async () => {
```

Comment:
You instantly forward to the new page, but the gebouwen.jsx page also contains a lot of html code. What is the point of that code? Also when no buildings are in the database, this will crash.

Reply:
The html things are to provide a bit of a placeholder while loading the buildings.
I'll have to fix the crash though